src/native/aws_signing.c

```diff
@@ -142,6 +142,7 @@ static void s_aws_signing_complete(struct aws_signing_result *result, int error_
 
     (*env)->CallBooleanMethod(
         env, callback_data->java_future, completable_future_properties.complete_method_id, java_signed_request);
+    AWS_FATAL_ASSERT(!(*env)->ExceptionCheck(env));
```

Comment:
why not:

AWS_FATAL_ASSERT(!aws_jni_check_and_clear_exception(env));

?

Reply:
I will switch all of them to that when your branch merges, that only exists in your branch ATM.